modules/wallet/encrypt.go

```diff
@@ -120,14 +130,30 @@ func (w *Wallet) unlock(masterKey crypto.TwofishKey) error {
 	return nil
 }
 
+// wipeSecrets erases all of the seeds and secret keys in the wallet.
+func (w *Wallet) wipeSecrets() {
+	// 'for i := range' must be used to prevent copies of secret data from
+	// being made.
```

Comment:
you never actually drop keys from the wallet. When you unlock the wallet, all the keys that existed previously will be initialized again. So there's no reason to clean out the map entirely, you just need to wipe the secret keys.